### PR TITLE
Shape check in `==` and `!=`

### PR DESCRIPTION
**Pull request by t-ae**
Resolved TODO in `==` and `!=`.

> // TODO: This is not correct due to broadcasting.

IMO, it doesn't make sense `[1, 2] == [[1, 2], [1, 2]]` is `true`, so I simply disabled broadcasting.

There was some test codes don't check equality strictly. Also fixed them.

### CODE REVIEW

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -282,23 +282,23 @@ final class LayerTests: XCTestCase {
         let layer = GlobalMaxPool1D<Float>()
         let input = Tensor(shape: [1, 10, 1], scalars: (0..<10).map(Float.init))
         let output = layer.inferring(from: input)
-        let expected = Tensor<Float>([9])
+        let expected = Tensor<Float>([[9]])
```
Comment (Shashi456) [inline Tests/TensorFlowTests/LayerTests.swift]: These are unrelated changes wrt to this PR, Are they absolutely needed? Because the tests passed before the change as well. But if it's still necessary I'd suggest opening a separate PR since they are layer test fixes.

File: Tests/TensorFlowTests/TensorAutoDiffTests.swift [hunk 2]
```diff
@@ -46,7 +46,7 @@ final class TensorAutoDiffTests: XCTestCase {
         func negate<T : TensorFlowFloatingPoint>(_ x: Tensor<T>) -> Tensor<T> {
             return (1 - x).sum()
         }
-        XCTAssertEqual(gradient(at: Tensor([0.1, 0.2, 0.3]), in: negate), Tensor(-1))
+        XCTAssertEqual(gradient(at: Tensor([0.1, 0.2, 0.3]), in: negate), Tensor([-1, -1, -1]))
```
Comment (Shashi456) [inline Tests/TensorFlowTests/TensorAutoDiffTests.swift]: Afaik, broadcasting is at works here.

File: Tests/TensorFlowTests/LayerTests.swift (hunk 1 above)
Comment (t-ae) [reply Tests/TensorFlowTests/LayerTests.swift]: `GlobalMaxPool1D` outputs tensor which has shape [BatchSize, Channels].
In here input shape is `[1, 10, 1]` and output shape is `[1, 1]`.
I disabled broadcasting in this PR so this test doesn't pass without change.

File: Tests/TensorFlowTests/TensorAutoDiffTests.swift (hunk 2 above)
Comment (t-ae) [reply Tests/TensorFlowTests/TensorAutoDiffTests.swift]: The TODO message originally written in `==` is:

> // TODO: This is not correct due to broadcasting.

It make sense to disable broadcastring. So I disabled it in this PR.

Comment (t-ae) [reply Tests/TensorFlowTests/TensorAutoDiffTests.swift]: My first PR comment didn't described the change enough.
I updated it.

Comment (Shashi456) [reply Tests/TensorFlowTests/TensorAutoDiffTests.swift]: In this specific case, since its a 1-D broadcast, I thought it was okay. The case you mentioned in the original comment though makes sense.
